### DISCUSSION TITLE
Executing scripts in @script

### DISCUSSION THREAD

**Discussion by ilyam2014** [livewire/livewire#7348] [1. Help]
Should it work in livewire components? because currently it doesn't
![image](https://github.com/livewire/livewire/assets/89526742/234c2ca0-85eb-4d30-b4e8-32ffbff30794)

**Comment by calebdw** [comment]
Everything needs to be an  a root div, does it not?

**Comment by sandersyao** [comment]
Maybe another problem. I use the script and endscript directive like this:

```blade
<div>
    {{-- 空组件，专门用于默认动态组件 --}}
</div>
@script
<script>
console.info("blank component loaded.");
</script>
@endscript
```

And then display script and endscript directive to web page.
<img width="614" alt="image" src="https://github.com/livewire/livewire/assets/2403124/0affe859-e0f6-44a4-9fda-bc2cc4f20335">

**Reply by ilyam2014** [reply]
afaik it only works in Livewire components maybe that's your problem

**Reply by sandersyao** [reply]
I think it maybe a livewire component:

```blade
{{-- In flow-editor.blade.php --}}
    <div class="w-full h-1/4">
{{-- Use blank component for default  --}}
      <livewire:is :component="$currentForm" :key="$currentForm" />
    </div>
```

```php
// In FlowEditor.php

class FlowEditor extends Component
{
    public string $currentForm = 'blank';
}
```

livewire version: 3.1.0

php version: 8.2.12

laravel version: 10.31.0

**Comment by Sparclex** [comment]
I've got the same issue.
Here is a simple reproduction: https://wirebox.app/b/xgrrk which results in the following error

![image](https://github.com/livewire/livewire/assets/2805213/3860bc2c-770d-4299-9896-16a5c2ca32c8)

**Reply by joshhanley** [reply]
@Sparclex can you update to the latest version of Livewire v3.2.6 and try again? As there has been some fixes done to `@assets` and `@script`

**Reply by ilyam2014** [reply]
https://wirebox.app/b/4n5l3
same

**Reply by joshhanley** [reply]
@hiddenLadder thanks for testing. I was hoping the latest changes would have sorted it. 😕 Any chance you could submit a failing test PR? Thanks!

**Reply by ilyam2014** [reply]
don't know how

**Reply by Sparclex** [reply]
created in https://github.com/livewire/livewire/pull/7508

**Reply by joshhanley** [reply]
@Sparclex thanks!

**Comment by rstanislav** [comment]
x-on:click is also not working, opened new, because this is closed: https://github.com/livewire/livewire/discussions/7789

**Comment by calebporzio** [comment]
I don't think Livewire's `@script` directive has ever supported defining a global function and accessing it from something like `@click`. Right? If I'm wrong, that wasn't intended to be supported, I don't think it's documented.

Anyhow, it would of course be awesome to support that, just letting you know I didn't have intentions of supporting it at least right away.

I'd love a failing test PR for the feature, if not an implementation. Thanks!

**Reply by rstanislav** [reply]
It would be great if there will be some way to use global functions with @x-on, maybe something like x-on:click.global - it will add alot of flexibility 😊

**Reply by rstanislav** [reply]
Alpine.data adds a lot of unnecessary code, and there is not much doc's\info about alpine on the internet still, and also there is still many v2 projects where it was ok to use @this.get to access livewire props in js global scope, i dont think its a great idea to use it in v3 since its not official supported way, and $wire on other hand is not avail outside script tags. Its possible to access it with Livewire.first or Livewire.getByName etc.. but thats 'dirty' way, that can lead to unpredictable results in future when any project grows and more components added.

**Reply by lucasweaver10** [reply]
@calebporzio When I read:

"Executing scripts
To execute bespoke JavaScript in your Livewire component, simply wrap a <script> element with @script and @endscript. This will tell Livewire to handle the execution of this JavaScript." -- https://livewire.laravel.com/docs/javascript#executing-scripts

I think that means we can write bespoke javascript functions in a <script> tag and then access $wire if we wrap it in @script. 

I'm not advocating that you add that functionality, since I've only needed it for one very specific edge case, just noting that maybe the docs could be updated to reflect that this isn't possible (without a custom workaround).
